src/ucs/sys/topo/base/topo.c

```diff
@@ -324,6 +324,49 @@ const char *ucs_topo_distance_str(const ucs_sys_dev_distance_t *distance,
     return ucs_string_buffer_cstr(&strb);
 }
 
+ucs_sys_device_t ucs_topo_set_sys_dev(const char *dev_name,
+                                      const char *sysfs_path,
+                                      int is_primary)
+{
+    const char *bdf_name;
+    ucs_status_t status;
+    ucs_sys_device_t sys_dev;
+    int num_devices_pre, override_name;
+
+    if (sysfs_path == NULL) {
+        goto out_unknown;
+    }
+
+    bdf_name = strrchr(sysfs_path, '/');
+    if (bdf_name == NULL) {
+        goto out_unknown;
+    }
+
+    ++bdf_name; /* Move past '/' separator */
+
+    /* Store number of devices before we add our device */
+    num_devices_pre = ucs_topo_num_devices();
+
+    status = ucs_topo_find_device_by_bdf_name(bdf_name, &sys_dev);
+    if (status != UCS_OK) {
+        goto out_unknown;
+    }
+
+    /* Override device name if device is new (number of devices increased),
+       or if it's a primary device */
+    override_name = (num_devices_pre < ucs_topo_num_devices()) || is_primary;
```

Comment:
IMO ucs_topo_num_devices check is not needed
what if just pass is_primary as override_name?

Reply:
what about TCP devices which are non-IB? (eg. ens32)
They will not get a name at all.

Reply:
Why not? if the device is new, it will use ens32 (or ib0 for example)
if mlx* device comes along with same id - it will override.
if mlx* device comes first, then ib0 will have override_name=0 so mlx* name will stay.